### PR TITLE
Delay user instruction in dockerfile generation

### PR DESCRIPTION
**Pull request by schustmi**
## Describe changes
This PR delays the `USER` instructions when generating dockerfiles and makes sure the specified user is the owner of the directory.

Adding the `USER` instruction early leads to some permission issues when installing pip packages on a system level. 
## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/utils/pipeline_docker_image_builder.py [hunk 1]
```diff
@@ -474,6 +471,10 @@ def _generate_zenml_pipeline_dockerfile(
 
         lines.append("RUN chmod -R a+rw .")
 
+        if docker_settings.user:
+            lines.append(f"USER {docker_settings.user}")
+            lines.append(f"RUN chown -R {docker_settings.user} .")
```
Comment (fa9r) [inline src/zenml/utils/pipeline_docker_image_builder.py:476]: Why is this necessary? Seems like this is the meat of the PR :D

Comment (schustmi) [reply src/zenml/utils/pipeline_docker_image_builder.py:476]: This user might not have the necessary permissions to install the `pip` packages in the above commands.
The `chown` command makes sure that the `/app` directory (which contains the user files/code) is owned by the provided user instead of `root`

**Review by fa9r — APPROVED** [review]
